src/lib/currency/currency.ml

```diff
@@ -66,6 +66,32 @@ end = struct
   [%%define_locally
   Unsigned.(to_uint64, of_uint64, of_int, to_int, of_string, to_string)]
 
+  let precision = 9
+
+  let precision_exp = Unsigned.of_int 1_000_000_000
+
+  let to_formatted_string amount =
```

Comment:
is `precision_exp` derivable from `precision`?

should these be fixed, or maybe they can be optional arguments to `of...` and `to...` with defaults ??

Reply:
I think they should be fixed since we've committed to these values in the whitepaper, but agree that they can be computed from one another.

Maybe something like:

```ocaml
let precision = 9
let precision_exp = 
     let open Unsigned in
     let go x i =
		if equal i zero then
           x
        else
           go (x * 10) (i-1) in go one (of_int precision))
```

Reply:
I think this will do:
```ocaml
let precision_exp = Unsigned.of_int @@ Int.pow 10 precision
```